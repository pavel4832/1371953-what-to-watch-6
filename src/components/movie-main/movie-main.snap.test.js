import React from 'react';
import {Provider} from 'react-redux';
import {render} from '@testing-library/react';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import MovieMain from './movie-main';
import configureStore from 'redux-mock-store';
import MOCK_STORE from '../../mock/mock-store';

const mockStore = configureStore({});

it(`Should MovieMain render correctly`, () => {
  const store = mockStore(MOCK_STORE);
  const history = createMemoryHistory();

  const {container} = render(
      <Provider store={store}>
        <Router history={history}>
          <MovieMain
            onPlayButtonClick={jest.fn()}
          />
        </Router>
      </Provider>
  );

  expect(container).toMatchSnapshot();
});
