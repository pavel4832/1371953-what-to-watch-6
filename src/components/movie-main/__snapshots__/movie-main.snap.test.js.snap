// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MovieMain render correctly 1`] = `
<div>
  <section
    class="movie-card"
  >
    <div
      class="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      class="visually-hidden"
    >
      WTW
    </h1>
    <header
      class="page-header movie-card__head "
    >
      <div
        class="logo"
      >
        <a
          class="logo__link"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      
      <h1
        class="page-title user-page__title"
        style="display: none;"
      />
      <div
        class="user-block"
      >
        <div
          class="user-block__avatar"
        >
          <a
            href="/mylist"
          >
            <img
              alt="User avatar"
              height="63"
              src="img/avatar.jpg"
              width="63"
            />
          </a>
        </div>
        <a
          class="user-block__link"
          href="/"
        >
          Log out
        </a>
      </div>
    </header>
    <div
      class="movie-card__wrap"
    >
      <div
        class="movie-card__info"
      >
        <div
          class="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          class="movie-card__desc"
        >
          <h2
            class="movie-card__title"
          >
            The Grand Budapest Hotel
          </h2>
          <p
            class="movie-card__meta"
          >
            <span
              class="movie-card__genre"
            >
              Dramas
            </span>
            <span
              class="movie-card__year"
            >
              2014
            </span>
          </p>
          <div
            class="movie-card__buttons"
          >
            <button
              class="btn btn--play movie-card__button"
              data-testid="play-button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlink:href="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              class="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  data-testid="add"
                  xlink:href="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
